src/buildercore/trop.py

```diff
@@ -574,24 +602,25 @@ def render_cloudfront(context, template, origin_hostname):
                 OriginProtocolPolicy='https-only' if context['cloudfront']['errors']['protocol'] == 'https' else 'http-only'
             )
         ))
-        props['CacheBehaviors'] = [
-            cloudfront.CacheBehavior(
-                TargetOriginId=CLOUDFRONT_ERROR_ORIGIN_ID,
-                DefaultTTL=context['cloudfront']['default-ttl'],
-                ForwardedValues=cloudfront.ForwardedValues(
-                    QueryString=False
-                ),
-                PathPattern=context['cloudfront']['errors']['pattern'],
-                ViewerProtocolPolicy='allow-all',
-            ),
-        ]
+        props['CacheBehaviors'] = [_cache_behavior(
+            CLOUDFRONT_ERROR_ORIGIN_ID,
+            context['cloudfront']['errors']['pattern'],
+        )]
         props['CustomErrorResponses'] = [
             cloudfront.CustomErrorResponse(
                 ErrorCode=code,
                 ResponseCode=code,
                 ResponsePagePath=page
             ) for code, page in context['cloudfront']['errors']['codes'].iteritems()
         ]
+
+    if context['cloudfront']['origins']:
+        props['CacheBehaviors'] = [
+            _cache_behavior(o_id, o['pattern'])
+            for o_id, o in context['cloudfront']['origins'].iteritems()
+            if o['pattern']
+        ]
```

Comment:
ha - the long-form list comprehension again. who knows, maybe I'll get used to it.